src/main/frontend/handler/page.cljs

```diff
@@ -200,26 +200,50 @@
      (p/catch (fn [error]
                 (println "file rename failed: " error))))))
 
+(defn- replace-page-ref!
+  [content old-name new-name]
+  (let [[original-old-name original-new-name] (map string/trim [old-name new-name])
+        [old-ref new-ref] (map #(util/format "[[%s]]" %) [old-name new-name])
+        [old-name new-name] (map #(when (string/includes? % "/")
+                                    (string/replace % "/" "."))
+                                 [original-old-name original-new-name])
+        old-org-ref (and (= :org (state/get-preferred-format))
+                         (:org-mode/insert-file-link? (state/get-config))
+                         (re-find
+                          (re-pattern
+                           (util/format
+                            "\\[\\[file:\\./.*%s\\.org\\]\\[(.*?)\\]\\]" old-name))
+                          content))]
+    (-> (if old-org-ref
+            (let [[old-full-ref old-label] old-org-ref
+                  new-label (if (= old-label original-old-name)
+                                original-new-name
+                              old-label)
+                  new-full-ref (-> (string/replace old-full-ref old-name new-name)
+                                   (string/replace (str "[" old-label "]")
+                                                   (str "[" new-label "]")))]
+              (string/replace content old-full-ref new-full-ref))
+          content)
+        (string/replace old-ref new-ref))))
+
+(defn- replace-tag-ref!
+  [content old-name new-name]
+  (let [old-tag (str " #" old-name)
```

Comment:
What's the purpose of the space before `#` here?

Reply:
Like abc#bar should not be treated as a page tag, which in this situation it will not be updated, even if bar is a existing page. How do you think? 

BTW the hash tag replacement does not take care of all the cases, it definitely needs more polish.

Reply:
It makes sense to me now. It seems that a tag at the beginning of a file might be a case that needs to be resolved. For example:
```
#tag 
```

Reply:
That’s the case we need to cover.